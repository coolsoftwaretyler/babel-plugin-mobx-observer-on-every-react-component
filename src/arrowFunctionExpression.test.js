import { expect, test, describe } from "bun:test";
import { transform } from "@babel/core";
import pluginSyntaxJsx from "@babel/plugin-syntax-jsx";
import pluginTransReactJsx from "@babel/plugin-transform-react-jsx";
import autoObserverPlugin from "./index.ts";

const runTransform = (input) => {
  return transform(input, {
    plugins: [pluginSyntaxJsx, pluginTransReactJsx, autoObserverPlugin],
  });
};

describe("Mobx Observer Babel Plugin", () => {
  describe("arrow function expression", () => {
    describe("when the function is not a react component", () => {
      test("it does not wrap the component in observer", () => {
        const source = `const MyArrow = () => 0`;
        const out = runTransform(source);

        expect(out.code).toMatchSnapshot();
      });
    });
    describe("when the function is already wrapped", () => {
      test("it does not wrap the component in observer", () => {
        const source = `const MyArrowObserver = observer(() => <div>Hello World</div>)`;
        const out = runTransform(source);

        expect(out.code).toMatchSnapshot();
      });
    });
    describe("when the function is not already wrapped", () => {
      test("it wraps the component in observer", () => {
        const source = `const MyArrowObserver = () => <div>Hello World</div>`;
        const out = runTransform(source);

        expect(out.code).toMatchSnapshot();
      });
    });
  });
});
