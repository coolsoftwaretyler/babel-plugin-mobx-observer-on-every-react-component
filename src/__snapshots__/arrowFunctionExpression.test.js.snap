// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobx Observer Babel Plugin arrow function expression when the function is not a react component it does not wrap the component in observer 1`] = `"const MyArrow = () => 0;"`;

exports[`Mobx Observer Babel Plugin arrow function expression when the function is already wrapped it does not wrap the component in observer 1`] = `"const MyArrowObserver = observer(() => /*#__PURE__*/React.createElement("div", null, "Hello World"));"`;

exports[`Mobx Observer Babel Plugin arrow function expression when the function is not already wrapped it wraps the component in observer 1`] = `"const MyArrowObserver = observer(() => /*#__PURE__*/React.createElement("div", null, "Hello World"));"`;
